src/inmanta/parser/plyInmantaParser.py

```diff
@@ -847,6 +855,36 @@ def p_constant_string(p: YaccProduction) -> None:
     attach_lnr(p)
 
 
+def p_constant_fstring(p: YaccProduction) -> None:
+    "constant : FSTRING"
+    formatter = string.Formatter()
+
+    # formatter.parse returns an iterable of tuple (literal_text, field_name, format_spec, conversion)
+    parsed: Iterable[Tuple[str, Optional[str], Optional[str], Optional[str]]] = formatter.parse(str(p[1]))
+
+    start_lnr = p[1].location.lnr
+    start_char_pos = p[1].location.start_char + 2  # FSTRING tokens begin with `f"` or `f'` of length 2
+
+    locatable_matches: List[Tuple[str, LocatableString]] = []
+    for match in parsed:
```

Comment:
I would add some testcase to check this line numbering logic is correct. 

(it is easy to test here, and very annoying in vscode if it is off)

cfr test_parser

Reply:
Added a `test_fstring_numbering_logic` test